solr/solrj/src/java/org/apache/solr/client/solrj/impl/BaseHttpClusterStateProvider.java

```diff
@@ -229,10 +236,9 @@ > getCacheTimeout()) {
       for (String nodeName : liveNodes) {
```

Comment:
If we exhaust liveNodes, shouldn't we then try the initial configured nodes, and then only failing that, throw an exception?